packages/extension-commerce-bm-seo/src/components/seo-hoc.tsx

```diff
@@ -29,18 +29,43 @@ const getComponent = (
     return ComponentClass
 }
 
+const CATCH_ALL_PATH = '*'
+
+/**
+ * Checks whether the given URL path matches a predefined route defined in the application's routes config, excluding the catch-all route (e.g., path='*')
+ */
+const isRouteDefined = (routeToMatch: string, routes: Array<{path: string}>): boolean => {
+    // Exclude any catch-all (404) routes
+    const validRoutes = routes.filter((route) => route.path !== CATCH_ALL_PATH)
```

Comment:
I believe is `path` is undefined that it will be treated as a wild card. So we might want to account for that. 

Additionally I'm wondering if we are to create a warning if there happens to be more than 1 invalid/wildcard routes.

The reason why I say that is we want the developer to have some insight is things don't behave the way they expect them to.